### PR TITLE
Disable backspace/delete handling on iOS & macOS

### LINKED ISSUE #91365: [iOS] Japanese text gets replaced when typing after deleting a character when using physical keyboard
## Steps to Reproduce

I don't have a verification environment at hand, so this is the content of an inquiry from a user
This can be a problem with excessive deletion, even though the delete button is only pressed once

It happened in both versions 2.5.2 and 2.6.0-5.2.pre.

Here's a video I got from a user.
The 5~7 seconds is where the problem is.
https://user-images.githubusercontent.com/7422251/136192069-3fcdfa1b-28c6-4747-a034-189580e0cade.mp4

Minimal sample code
```
class SamplePage extends StatefulWidget {

  SamplePage(): super();

  @override
  SamplePageState createState() => SamplePageState();
}

class SamplePageState extends State<SamplePage> with WidgetsBindingObserver {

  @override
  void initState(){
    super.initState();
  }

  Widget build(BuildContext context){
    return Container(
      padding: EdgeInsets.only(top: 100, left: 10, right: 10),
      child: TextField(),
    );
  }
}

```

1. Run `flutter run`.
2. input japanese word. ( sample. "ご覧に" is good. In english keyboard input "goranni" )
3. press delete Button ( sentense will be "ご覧" )
4. input "に". In english keyboard input "ni". 

**Expected results:** 
the word will be "ご覧に"

**Actual results:** 
the word will be "に"

<details>

`flutter doctor -v`
```
[✓] Flutter (Channel stable, 2.5.2, on Mac OS X 10.15.7 19H1419 darwin-x64, locale ja-JP)
    • Flutter version 2.5.2 at /Users/user/Desktop/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (6 days ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3
[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/user/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install “cmdline-tools;latest”`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.
[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.4, Build version 12D4e
    • CocoaPods version 1.11.0
[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      :hammer: https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 94.0.4606.61
```

</details>

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
Fixes https://github.com/flutter/flutter/issues/91365, https://github.com/flutter/flutter/issues/111113, https://github.com/flutter/flutter/issues/96660
`DeleteToLineBreak` will continue to be handled by the framework since it's layout dependent.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart [hunk 1]
```diff
@@ -447,6 +453,19 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
     const SingleActivator(LogicalKeyboardKey.arrowUp, shift: true, alt: true): const DoNothingAndStopPropagationTextIntent(),
   };
 
+  // Hand backspace/delete events that do not depend on text layout back to the
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:456]: This comment is a bit confusing, maybe add some more explanation why this is needed.

Also, would this apply to macos?

**Review by chunhtai — APPROVED** [review]
LGTM, left some comments

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:456]: macOS already does that (and the default shortcuts are already disabled)

Comment (chunhtai) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: is that true? I saw the _macDisablingTextShortcuts is using _commonDisablingTextShortcuts which does not include delete and backward

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: huh backspacing isn't deleting composing text on macOS

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: added macOS delete/backspaces to the list. From the looks of it that fixes https://github.com/flutter/flutter/issues/111113

Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart]: The change seems to work fine with the default macOS chinese/japanese IME

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart [hunk 2]
```diff
@@ -440,13 +446,28 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
...
+    ..._iOSDisablingTextShortcuts,
     const SingleActivator(LogicalKeyboardKey.escape): const DoNothingAndStopPropagationTextIntent(),
     const SingleActivator(LogicalKeyboardKey.tab): const DoNothingAndStopPropagationTextIntent(),
     const SingleActivator(LogicalKeyboardKey.tab, shift: true): const DoNothingAndStopPropagationTextIntent(),
     const SingleActivator(LogicalKeyboardKey.arrowDown, shift: true, alt: true): const DoNothingAndStopPropagationTextIntent(),
     const SingleActivator(LogicalKeyboardKey.arrowUp, shift: true, alt: true): const DoNothingAndStopPropagationTextIntent(),
   };
 
+  // Hand backspace/delete events that do not depend on text layout (delete
+  // character and delete to the next word) back to the IME to allow it update
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:458]: "allow it" => "allow it to"

File: packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart [hunk 3]
```diff
@@ -19,6 +19,12 @@ import 'text_editing_intents.dart';
 /// lower in the widget tree than this. See the [Action] class for an example
 /// of remapping an [Intent] to a custom [Action].
 ///
+/// The [Shortcuts] widget usually takes precedence over system keybindings.
+/// Proceed with caution if the shortcut you wish to override is also used by
+/// the system. For example, overriding [LogicalKeyboardKey.backspace] could
+/// cause CJK input methods to discard more text than they should when the
+/// backspace key is pressed during text composition on iOS.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart:26]: Good call adding this to the docs.

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Does the problem with backspace not working in composing regions not happen on other platforms?

**Comment by LongCatIsLooong** [comment]
> Does the problem with backspace not working in composing regions not happen on other platforms?

Most android IMEs can deal with these changes I think (and most of them don't use composing regions for CJK). Not sure about Windows/Linux since I don't have the test environment. /cc @cbracken

**Comment by LongCatIsLooong** [comment]
On a side note, arrow key navigation within IME candidate window still doesn't work on iOS with an external keyboard. We may need to disable those as well in the future.

**Pull request merged**